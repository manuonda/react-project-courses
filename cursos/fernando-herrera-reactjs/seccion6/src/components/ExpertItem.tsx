import React from 'react'

type Props = {
    item: string;
}

export const ExpertItem:React.FC<Props> = ({ item }) => {
    return (
        <>
          <li>{item}</li>   
        </>
    )
}
