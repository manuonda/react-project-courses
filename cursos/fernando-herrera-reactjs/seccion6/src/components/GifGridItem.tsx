import React from 'react'
import { Gif } from '../types'
import './style.css';

type Props = {
    image: Gif
}

const GifGridItem:React.FC<Props> = ({ image }) => {
    return (
        <>
          <div className="card">
            <img src={image.url} alt={image.title}></img>    
            <p>{image.title}</p>
          </div> 
        </>
    )
}

export default GifGridItem
