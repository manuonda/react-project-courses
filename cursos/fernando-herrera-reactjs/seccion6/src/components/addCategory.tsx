import { CategorySharp } from "@material-ui/icons";
import React , {SetStateAction,ChangeEvent  ,FormEvent , Dispatch, useState} from "react"


/**
 * Add Category 
 */

 type inputChange = ChangeEvent<HTMLInputElement>;
  
 type Props = {
    handleCategorys: Dispatch<SetStateAction<string[]>> ;
 }

 const AddCategory:React.FC<Props> = ({handleCategorys}) => {
    const [value, setValue] = useState<string>("");
    const handleInputChange = (e : inputChange)  => {
       setValue(e.target.value);
    }

    const handleSubmit = (e:FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        handleCategorys((categorys:string[]) => [...categorys, value]);
        console.log(value);
        setValue("");
        console.log(value);

    }
    return(
        <form onSubmit={handleSubmit}>
             <input type="text" onChange={(e) => handleInputChange(e)}/> 
           <button type="submit">Agregar</button>  
        </form>
      )
}

export default AddCategory;