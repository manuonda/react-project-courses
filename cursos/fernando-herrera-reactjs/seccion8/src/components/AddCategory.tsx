import React, { useState } from 'react'
import PropTypes from 'prop-types';

type Props = {
    setCategories: any
}

export const AddCategory: React.FC<Props> = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e: React.ChangeEvent<HTMLInputElement> ) => {
        setInputValue( e.target.value );
        console.log('jhanldeInputChange llamado');
    }

    const handleSubmit = (e: any) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( (cats:any) => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
