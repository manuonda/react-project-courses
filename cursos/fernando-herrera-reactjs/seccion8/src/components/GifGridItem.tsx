import React from 'react'

type Props = {
    title: string,
    url: string
}

export const GifGridItem = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}
