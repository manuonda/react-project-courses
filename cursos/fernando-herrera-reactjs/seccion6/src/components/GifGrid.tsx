import React, {useState, useEffect} from 'react'
import {BASE_URL, KEY } from '../constants/index'
import { Gif , State} from '../types';
import GifGridItem from '../components/GifGridItem';

import { useFetchGifs }  from '../hooks/useFetchGifs';


type Props = {
    category:string;
}

const initState:State = {
   data:[],
   loading:false
}
export const ExpertItem:React.FC<Props> = ({ category }) => {

  
    const {data , loading }= useFetchGifs(category);
    //setState(result);
  
    return (
        <div className="card-grid">
            <h1>{category}</h1>
            <br></br>
          {loading && (
            <h1>Loading</h1>
           )} 
          { data && data.map((image:Gif) => (
            <GifGridItem key={image.id} image={image}/>
          ))}
        </div>
    )
}
