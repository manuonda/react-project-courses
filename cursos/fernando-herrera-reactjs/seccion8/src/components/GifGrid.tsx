import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

type Props = {
    category: any
}

type Img = {
    title: string,
    url: string
}

export const GifGrid: React.FC<Props> = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> { category } </h3>

            { loading && <p className="animate__animated animate__flash">Loading</p> }

            <div className="card-grid">
                
                {
                    images.map( (img:any) => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            
            </div>
        </>
    )
}
