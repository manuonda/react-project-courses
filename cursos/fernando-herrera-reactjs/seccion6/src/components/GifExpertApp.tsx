import React , {useState} from 'react'
import PropTypes from 'prop-types'
import { ExpertItem } from './ExpertItem';
import AddCategory from './addCategory';

type Category = {
    nombre: string;
}
const categorias:string[] = ['One Puch Man', 'Samurai X'];
const GifExpertApp:React.FC = ( ) => {

    const [categorys, setCategorys] = useState<string[]>(categorias);
    
    const handleAdd = () => {
        setCategorys([
            ...categorys,
            "HUNTERX2"
        ]);
        setCategorys( cats => [...cats, "Hunberx"]);
        categorys.sort();
    }
    
    return (
        <>
        <AddCategory handleCategorys={setCategorys}></AddCategory>
          <button onClick={handleAdd}></button>
           {categorys.map((item:string) => (
             <ExpertItem key={item} item={item}>{item}</ExpertItem>
           ))}           
        </>
    )
}

GifExpertApp.propTypes = {

}

export default GifExpertApp
