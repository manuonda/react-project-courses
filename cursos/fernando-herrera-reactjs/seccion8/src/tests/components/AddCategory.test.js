import React from 'react'
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory";

describe('Pruebas en el compoentno <AddCategory>', () => {

    const setCategories = () => jest.fn();
    const wrapper = shallow( <AddCategory setCategories={setCategories}/>);
   

    test('debe de mostrarse correctamente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('debe de cambiar la caja de texto', () => {
       const input = wrapper.find('input');
       const value = "Hola mundo";
       // simula el change y envia el target
       input.simulate('change', { target: { value : value } })
   })

});