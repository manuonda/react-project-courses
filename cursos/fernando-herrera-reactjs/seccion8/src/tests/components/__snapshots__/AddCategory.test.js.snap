// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el compoentno <AddCategory> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
